internal/performance/performance.go

```diff
@@ -164,16 +176,34 @@ func (c *PerfCollector) processEntityMetrics(metricsValues *types.PerfEntityMetr
 
 		// This is a short-lived object, the purpose is to compute the average of the different performance metrics
 		// when more than one instance per entity returns a value
-		if _, ok := accumulateMetrics[name]; !ok {
-			accumulateMetrics[name] = &Accumulator{}
+		pe, ok := accumulateMetrics[name]
+		if !ok {
+			pe = &perfEvaluer{accumulator: accumulator{}}
+			accumulateMetrics[name] = pe
+		}
+
+		if metricValue.GetPerfMetricSeries().Id.Instance == "" {
+			pe.instancelessValue = &metricVal
+		} else {
+			pe.accumulator.Occurrences++
+			pe.accumulator.Sum += metricVal
 		}
-		accumulateMetrics[name].Occurrences++
-		accumulateMetrics[name].Sum += metricVal
+
 	}
+
 	for key, val := range accumulateMetrics {
+		var value int64
+
+		//We give priority to the raw values and fall back to 'instanceless' values in case no raw data has been received
+		if val.accumulator.Occurrences != 0 {
+			value = val.accumulator.Sum / val.accumulator.Occurrences
+		} else if val.instancelessValue != nil {
```

Comment:
i would switch the order here just to be consistent with the order above also to me it express a bit better which has precedence.